sources/refund.move

```diff
@@ -131,100 +208,171 @@ module refund::refund {
-        let available = balance::value(&pool.funds);
-        if (available >= pool.accounting.current_liability) {
-            0
-        } else {
-            pool.accounting.current_liability - available
-        }
+    
+    public fun get_total_to_refund(pool: &RefundPool): u64 { total_to_refund(&pool.accounting) }
+    public fun get_total_raised(pool: &RefundPool): u64 { total_raised(&pool.accounting) }
+    public fun get_total_refunded(pool: &RefundPool): u64 { total_refunded(&pool.accounting) }
+    public fun get_total_boosted(pool: &RefundPool): u64 { total_boosted(&pool.accounting) }
+    public fun base_funds(pool: &RefundPool): u64 { balance::value(funds(&pool.base_pool)) }
+    public fun booster_funds(pool: &RefundPool): u64 { balance::value(funds(&pool.booster_pool)) }
+    public fun current_liabilities(pool: &RefundPool): u64 { accounting::current_liabilities(&pool.accounting) }
+
+    // === Mutators (Friends) ===
+    
+    public(friend) fun unclaimed_mut(pool: &mut RefundPool): &mut Table<address, u64> { &mut pool.unclaimed }
+    public(friend) fun accounting_mut(pool: &mut RefundPool): &mut Accounting { &mut pool.accounting }
+    public(friend) fun booster_pool_mut(pool: &mut RefundPool): &mut Pool { &mut pool.base_pool }
+
+    // === Phase Transitions ===
+
+    public entry fun start_timeout_setup_phase(
+        pub: &Publisher,
+        pool: &mut RefundPool,
+    ) {
+        assert_address_addition_phase(pool);
+        assert_publisher(pub);
+
+        next_phase(pool)
     }
+    
+    public entry fun start_funding_phase(
+        pub: &Publisher,
+        pool: &mut RefundPool,
+    ) {
+        assert_timeout_setup_phase(pool);
+        assert!(is_some(&pool.timeout_ts), ETimeoutTimestampIsNone);
+        assert_publisher(pub);
 
-    /// Calculates the unfunded liability of the RefundPool under boosted refund conditions.
-    ///
-    /// Similar to `unfunded_liability`, but considers the scenario where all refunds are boosted by 50%.
-    /// This function first calculates the standard unfunded liability, then adds 50% of that liability to
-    /// model the additional funds that would be required if all refunds were claimed with a 150% boost.
-    public fun unfunded_liability_boosted(pool: &RefundPool): u64 {
-        let unfunded_liability = unfunded_liability(pool);
-        unfunded_liability + div(unfunded_liability, 2)
+        next_phase(pool)
     }
 
+    public entry fun start_claim_phase(
+        pool: &mut RefundPool,
+    ) {
+        assert_funding_phase(pool);
+        let total_to_refund = total_to_refund(&pool.accounting);
+        let total_raised = total_raised(&pool.accounting);
+        assert!(total_to_refund == total_raised, EPoolUnderfunded);
+
+        next_phase(pool)
+    }
+    
+    public entry fun start_reclaim_phase(
+        pool: &mut RefundPool,
+        clock: &Clock,
+    ) {
+        assert_claim_phase(pool);
```

Comment:
I think there is one state transition not accounted for here: in the funding phase, but the fundraising target has not been hit, and the timeout has already been reached. I think in that case, it should be possible to advance from funding -> reclaim

Reply:
Implemented